sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -119,4 +119,13 @@ object IntervalUtils {
       case _: IllegalArgumentException => null
     }
   }
+
+  def legacyCastStringToInterval(str: String): CalendarInterval = {
+    val trimmed = str.trim
+    if (trimmed.startsWith("interval")) {
```

Comment:
I think this should be case insensitive check. You can use `regionMatches()` for that.